File: app/src/main/java/eu/caraus/dynamo/application/domain/udacity/AffiliatesItem.java
package eu.caraus.dynamo.application.domain.udacity;


import com.google.gson.annotations.SerializedName;

import java.io.Serializable;


public class AffiliatesItem implements Serializable {

	@SerializedName("image")
	private String image;

	@SerializedName("name")
	private String name;

	@SerializedName("key")
	private String key;

	public void setImage(String image){
		this.image = image;
	}

	public String getImage(){
		return image;
	}

	public void setName(String name){
		this.name = name;
	}

	public String getName(){
		return name;
	}

	public void setKey(String key){
		this.key = key;
	}

	public String getKey(){
		return key;
	}

	@Override
 	public String toString(){
		return 
			"AffiliatesItem{" + 
			"image = '" + image + '\'' + 
			",name = '" + name + '\'' + 
			",key = '" + key + '\'' + 
			"}";
		}
}

File: app/src/main/java/eu/caraus/dynamo/application/data/domain/udacity/AffiliatesItem.kt
package eu.caraus.dynamo.application.data.domain.udacity


import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey
import android.arch.persistence.room.Index
import android.arch.persistence.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import io.reactivex.annotations.NonNull

import java.io.Serializable

@Entity(

        tableName   = "affiliates_item",

        indices     = [( Index("key") )],

        foreignKeys = [( ForeignKey( entity = CoursesItem::class,
                                     parentColumns = ["key"],
                                     childColumns = ["key"],
                                     onDelete = ForeignKey.CASCADE )
                      )]
)
data class AffiliatesItem  (

   @NonNull
   @PrimaryKey
   @SerializedName("key")
   var key: String,

   @SerializedName("name")
   var name: String? = null,

   @SerializedName("image")
   var image: String? = null

) : Serializable {
  // constructor() : this("" ,"" ,"" )
}